### PR TITLE
Enable compiled-in XPM pixmaps (#4578)

### PR DESCRIPTION
**Pull request by JohannesLorenz**
Should I merge this to master? It's practical and may help coders in the future, but it's also currently unused code (as it's only inside my osc-plugin branch), so coders have no example about how it must be used.

### CODE REVIEW

**Comment by JohannesLorenz** [comment]
Note: CI only failed because of a used `assert()` without an include. I'll fix this soon (it looks like it's `Q_ASSERT` is common style?).

**Comment by PhysSong** [comment]
`Q_ASSERT` crashes the program if not built with `QT_NO_DEBUG`. Currently we don't define it in release builds so `Q_ASSERT` will always crash LMMS.

**Comment by JohannesLorenz** [comment]
@PhysSong  Thanks for the hint. Aborting if a pixmap is not loaded sounds like overkill, even in debug mode. I changed it into a warning now in e0384b8 , and the code will safely run on ...

File: src/gui/embed.cpp [hunk 1]
```diff
@@ -52,6 +52,15 @@ QPixmap getIconPixmap(const QString& pixmapName, int width, int height )
 	}
 	QImageReader reader(QString("artwork:%1").arg(pixmapName));
 
+	// xpm strings are not loaded from file, you must use addXpmToCache()
+	if (pixmapName.startsWith("xpm:"))
```
Comment (PhysSong) [inline src/gui/embed.cpp]: I think it'd be better if this check were before creating a `QImageReader` instance.

File: include/embed.h [hunk 2]
```diff
@@ -35,6 +36,32 @@
...
+ * Once done, you can use getIconPixmap() or new PixmapLoader() with
+ * "xpm:<unique identifier>".
+ *
+ * @note This function can only be called while a GuiApplication is running.
+ *   Otherwise, expect segmentation faults.
+ *
+ * @param pixmapName Unique identifier for the XPM. Do *not* prepend "xpm:".
+ * @param xpm_data The raw XPM string (*not* a file name).
+ */
+inline void addXpmToCache(const QString& pixmapName, const char** xpm_data) {
```
Comment (PhysSong) [inline include/embed.h]: > This function can only be called while a `GuiApplication` is running.

Can we check it in this function and warn if the condition not met?

**Pull request merged**